### PR TITLE
feat: Open edX app theming capability improvements

### PR DESCRIPTION
**Pull request by omerhabib26**
### How we can update the theme:-
Steps to be followed:- 
1. Create a resource directory that needs to be customized i.e., Fonts and Resources.
   - Path: `core/src/{directory_name}/res/`
   - Add only those resources that need to be customized i.e, `logo, symbols or icons`
   - Place the `font.xml` file in the `src/{directory_name}/res/font` directory, alongside the `font files`.
2. Update the resource path from `core/build.gradle`.
    ```
    sourceSets {
       {enviorment_name} {
            res.srcDirs = ['src/{directory_name}/res']
        }
    }
    ```
3. Update the `Colors.kt` according to the desired colors.

### Pre-requisite:-  
- We need to move colors codes from `theme.kt` to a separate `Colors.kt` as per material design recommendations. We can also follow the naming conventions suggested by the [Material Design 3](https://m3.material.io/styles/color/roles).
- We should specify fonts for the application rather than relying on the device's default fonts, and it's essential to update the font references in Type.kt.
   - It's advisable to include certain default fonts within the codebase.
   
   
PS: As we are going for new architecture we should migrate to Material.io 3.

### CODE REVIEW

File: core/build.gradle [hunk 1]
```diff
@@ -103,6 +103,12 @@ android {
         }
     }
 
+    sourceSets {
+        develop {
+            res.srcDirs = ['src/edx/res']
```
[the remark this answers is not in the document]
Comment (volodymyr-chekyrta) [reply core/build.gradle:108]: By the way, it seems we just don't need the edx folder for the Open edX project.
Was it added intentionally or for demonstration purposes?

image

[the remark this answers is not in the document]
Comment (omerhabib26) [reply core/build.gradle:108]: as the current implementation was just for reference I'm updating the PR as we discussed in tech-discussion.

[the remark this answers is not in the document]
Comment (omerhabib26) [reply core/build.gradle]: @volodymyr-chekyrta PR updated, please use the latest code

**Review by k1rill — APPROVED** [review]

**Comment by volodymyr-chekyrta** [comment]
Looks good to me 👍.
It's a draft. Should we merge or expect more in this PR?

**Comment by omerhabib26** [comment]
> Looks good to me 👍. It's a draft. Should we merge or expect more in this PR?

we can merge it now. 🚀

**Pull request merged**